packages/shared-components/src/components/onto-repository-selector/onto-repository-selector-button/onto-repository-selector-button.tsx

```diff
@@ -0,0 +1,71 @@
+import {Component, Host, h, Prop, State} from '@stencil/core';
+import {Repository, UriUtil} from '@ontotext/workbench-api';
+import {TranslationService} from '../../../services/translation.service';
+
+@Component({
+  tag: 'onto-repository-selector-button',
+  styleUrl: 'onto-repository-selector-button.scss',
+  shadow: false,
+})
+/**
+ *
+ * A StencilJS component that renders a repository selector button. The button displays the repository's icon and ID
+ * if a repository is selected; otherwise, it shows a translated toggle button name. The component dynamically
+ * updates the button name when the translation changes.
+ *
+ * @example
+ * <onto-repository-selector-button repository={repository}></onto-repository-selector-button>
+ *
+ */
+ export class OntoRepositorySelectorButton {
```

Comment:
Not sure if this must be a standard component and not a functional one?! If the state is managed in the selector component as it should be IMO, this component can clearly be a functional component.

Reply:
Refactored to functional component